include/unifex/let_with.hpp

```diff
@@ -0,0 +1,134 @@
+
+    template<template<typename...> class Variant>
+    using error_types = typename InnerOp::template error_types<Variant>;
+
+    template<typename StateFactory2, typename SuccessorFactory2>
+    explicit type(StateFactory2&& stateFactory, SuccessorFactory2&& func) :
+        stateFactory_((StateFactory2&&)stateFactory),
+        func_((SuccessorFactory2&&)func)
+    {}
+
+    template(typename Self, typename Receiver)
+        (requires same_as<remove_cvref_t<Self>, type>)
+    friend auto tag_invoke(tag_t<unifex::connect>, Self&& self, Receiver&& r)
+        noexcept(
+            std::is_nothrow_invocable_v<member_t<Self, StateFactory>> &&
+            std::is_nothrow_invocable_v<
+                member_t<Self, SuccessorFactory>,
+                std::invoke_result_t<member_t<Self, StateFactory>>&> &&
+            std::is_nothrow_invocable_v<
+                decltype(unifex::connect),
+                std::invoke_result_t<
+                    member_t<Self, SuccessorFactory>,
+                    std::invoke_result_t<member_t<Self, StateFactory>>&>,
+                remove_cvref_t<Receiver>>) {
+
+        return operation<
+                member_t<Self, StateFactory>, member_t<Self, SuccessorFactory>, Receiver>(
+            static_cast<Self&&>(self).stateFactory_,
+            static_cast<Self&&>(self).func_,
+            static_cast<Receiver&&>(r));
+    }
+
+private:
+    UNIFEX_NO_UNIQUE_ADDRESS StateFactory stateFactory_;
+    UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
+};
+
+
+template<typename StateFactory, typename SuccessorFactory, typename Receiver>
+struct _let_with_operation<StateFactory, SuccessorFactory, Receiver>::type {
+    type(StateFactory&& stateFactory, SuccessorFactory&& func, Receiver&& r) :
+        state_(static_cast<StateFactory&&>(stateFactory)()),
+        innerOp_(
+              unifex::connect(
+                static_cast<SuccessorFactory&&>(func)(state_),
+                static_cast<Receiver&&>(r))) {
+    }
+
+    void start() & noexcept {
+        unifex::start(innerOp_);
+    }
+
+    std::invoke_result_t<StateFactory> state_;
+    connect_result_t<
+        std::invoke_result_t<SuccessorFactory, std::invoke_result_t<StateFactory>&>,
+        remove_cvref_t<Receiver>>
+        innerOp_;
+};
+
+struct _fn {
+
+    template<typename StateFactory, typename SuccessorFactory>
+    auto operator()(StateFactory&& stateFactory, SuccessorFactory&& successor_factory) const
+        noexcept(std::is_nothrow_constructible_v<remove_cvref_t<SuccessorFactory>, SuccessorFactory> &&
+                 std::is_nothrow_constructible_v<remove_cvref_t<StateFactory>, StateFactory>)
+        -> let_with_sender<remove_cvref_t<StateFactory>, remove_cvref_t<SuccessorFactory>> {
+        return let_with_sender<remove_cvref_t<StateFactory>, remove_cvref_t<SuccessorFactory>>{
+            (StateFactory&&)stateFactory, (SuccessorFactory&&)successor_factory};
+    }
+};
```

Comment:
It's best from an ADL-isolation point of view to put the CPO types into their own namespace. Since they get used in ADL calls, we don't want them accidentally dragging in any overloads that might happen to be in its associated namespace. (This matters more on MSVC which erroneously dumps hidden friend functions into the enclosing namespace.)